src/wasm/src/lib.rs

```diff
@@ -32,3 +48,235 @@ impl WasmLockfile {
+      }
+      MappedResolution::PackageJson {
+        dep_result,
+        sub_path,
+        ..
+      } => {
+        let result = dep_result
+          .as_ref()
+          .map_err(|e| JsError::new(&e.to_string()))?;
+
+        match result {
+          PackageJsonDepValue::Workspace { .. } => {
+            Err(JsError::new("Workspace dep is not supported"))
+          }
+          PackageJsonDepValue::Req(req) => Ok(format!(
+            "npm:{}@{}{}",
+            req.name,
+            req.version_req,
+            sub_path.map(|p| format!("/{}", p)).unwrap_or_default()
+          )),
+        }
+      }
+    }
+  }
+}
+
+#[wasm_bindgen(module = "/fs.js")]
+extern "C" {
+  #[wasm_bindgen(catch)]
+  fn stat_sync(path: &str) -> Result<JsValue, JsValue>;
+  #[wasm_bindgen(catch)]
+  fn read_to_string_lossy(path: &str) -> Result<String, JsValue>;
+  #[wasm_bindgen(catch)]
+  fn read_dir(path: &str) -> Result<JsValue, JsValue>;
+}
+
+#[derive(Debug, Default, Clone, Deserialize)]
+pub struct WasmFsMetadata {
+  pub is_file: bool,
+  pub is_directory: bool,
+  pub is_symlink: bool,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct WasmFsDirEntry {
+  pub name: String,
+  #[serde(flatten)]
+  pub metadata: WasmFsMetadata,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Error {
+  pub message: String,
+  pub code: Option<String>,
+}
+
+struct WasmFs;
+
+impl DenoConfigFs for WasmFs {
+  fn stat_sync(
+    &self,
+    path: &std::path::Path,
+  ) -> Result<FsMetadata, std::io::Error> {
+    let path = path.to_string_lossy().to_string();
+    let value = stat_sync(&path).map_err(map_err)?;
+    let metadata: WasmFsMetadata =
+      serde_wasm_bindgen::from_value(value).unwrap();
+    Ok(FsMetadata {
+      is_file: metadata.is_file,
+      is_directory: metadata.is_directory,
+      is_symlink: metadata.is_symlink,
+    })
+  }
+
+  fn read_to_string_lossy(
+    &self,
+    path: &std::path::Path,
+  ) -> Result<String, std::io::Error> {
+    let path = path.to_string_lossy().to_string();
+    let value = read_to_string_lossy(&path).map_err(map_err)?;
+    Ok(value)
+  }
+
+  fn read_dir(
+    &self,
+    path: &std::path::Path,
+  ) -> Result<Vec<deno_config::fs::FsDirEntry>, std::io::Error> {
+    let path2 = path.to_string_lossy().to_string();
+    let bytes = read_dir(&path2).map_err(map_err)?;
+    let entries: Vec<WasmFsDirEntry> =
+      serde_wasm_bindgen::from_value(bytes).unwrap();
+    Ok(
+      entries
+        .into_iter()
+        .map(|entry| deno_config::fs::FsDirEntry {
+          path: path.join(entry.name),
+          metadata: FsMetadata {
+            is_file: entry.metadata.is_file,
+            is_directory: entry.metadata.is_directory,
+            is_symlink: entry.metadata.is_symlink,
+          },
+        })
+        .collect(),
+    )
+  }
+}
+
+fn map_err(err: JsValue) -> std::io::Error {
+  let error: Error = serde_wasm_bindgen::from_value(err).unwrap();
```

Comment:
Maybe this should handle scenarios where it's not of type `Error`? Probably won't happen though.

Reply:
Eh, I think it won't happen.